tools/tagging.py

```diff
@@ -0,0 +1,66 @@
+import os
+import subprocess
+import sys
+
+from tools import ASAPP_ROOT, ASAPP_MLENG_ROOT
+from tools import CLIENT_FULL_NAMES
+
+import pandas as pd
+
+
+class ProcessTagsThatClientReturns:
+
+    EXPECTED_INPUT_FILE_COLUMNS = ['text', 'tag', 'notes']
+
+    def __init__(self, config):
+        self._config = config['tools']['tagging']
+        self._client = config['client']
+        self._start_date = config['start_date']
+        self._end_date = config['end_date']
+        self._input_directory = os.path.join(ASAPP_ROOT, 'data', self._client, self._start_date)
+
+    @property
+    def input_file(self):
+        return os.path.join(
+            self._input_directory,
+            f'{CLIENT_FULL_NAMES[self._client]}-tagsource{self._start_date}.csv'
+        )
+
+    def run(self):
```

Comment:
if run is almost always going to run _validate_input() and _run_steps() it should be implemented in the base class